arm_robots/src/arm_robots/robot.py

```diff
@@ -89,7 +97,7 @@ def set_execute(self, execute: bool):
     def set_block(self, block: bool):
         self.block = block
 
-    def get_move_group_commander(self, group_name: str) -> moveit_commander.MoveGroupCommander:
+    def _get_move_group_commander(self, group_name: str) -> moveit_commander.MoveGroupCommander:
```

Comment:
`_get_move_group_commander` is only called once, meaning that setting `self.max_velocity_scale_factor` will (silently and unexpectedly) not actually change the planned max velocity.

We discussed hypothetical issues with stored `move_group_commander`s not updating from changes in ROS, and this framework is okay there. However this framework is not okay for the very practical scenarios of changing `max_velocity_scale_factor`.

I'll make a PR with my solution to your problem